### DISCUSSION TITLE
Unable to Flash Beca BAC-6000

### DISCUSSION THREAD

**Discussion by mojtabavi** [arendst/Tasmota#17241] [Q&A] [answered]
I'm Trying to flash beca BAC-6000 thermostat using CH340G device
But the desired esp module is not placed in the program mode and sends incomprehensible characters on the serial
ESP Model: 
TYWE3S
Connection:
esp-gnd --> gnd
esp-3v3 --> 3v3
esp-gpio0 --> gnd
esp-rx --> tx
esp-tx --> rx
![Screenshot 2022-12-01 141155](https://user-images.githubusercontent.com/63472358/205034386-4607e6b7-bfc9-4da0-8999-7d33abf13c23.png)
![photo_2022-12-01_14-09-15](https://user-images.githubusercontent.com/63472358/205034639-79a4ef04-0aca-4979-bc9d-9215a03da120.jpg)
I'm really confused, I'd really appreciate it if someone could help

**Comment by mojtabavi** [comment]
https://user-images.githubusercontent.com/63472358/205036010-e8d3612c-2c84-4e5e-8e38-842dd2eebb89.mp4

**Comment by julred** [comment]
Correct me if I'm wrong - but this is not a tasmota related issue

**Comment by mojtabavi** [comment]
> Correct me if I'm wrong - but this is not a tasmota related issue

I know, but i'm trying to flash tasmota, And I'm trying to get help

**Comment by barbudor** [comment]
It is not a Tasmota Issue so indeed it should have been into Discussions instead of Issue

However @mojtabavi You don't show how you have tried to put the ESP in flashing mode
Are you 100% to have connected GPIO0 to GND ?
How ? Directly with a wire ? On the TYWES3 module ?
The connection must be done BEFORE applying power
I would recommand soldering a wire directly on the TYWES3 between GPIO and GND

**Comment by mojtabavi** [comment]
> It is not a Tasmota Issue so indeed it should have been into Discussions instead of Issue
> 
> However @mojtabavi You don't show how you have tried to put the ESP in flashing mode Are you 100% to have connected GPIO0 to GND ? How ? Directly with a wire ? On the TYWES3 module ? The connection must be done BEFORE applying power I would recommand soldering a wire directly on the TYWES3 between GPIO and GND

yes I soldering a wire directly on the TYWES3 between GPIO and GND

**Comment by barbudor** [comment]
Could be a deffective device but can't tell more
No photos of your actual wiring and setup no it could be anything

**Comment by joba-1** [comment]
Do you use baud rate 74880 during boot? Might make incomprehensible characters readable and give a hint on boot mode.

**Reply by mojtabavi** [reply]
> Do you use baud rate 74880 during boot? Might make incomprehensible characters readable and give a hint on boot mode.

![serial_74880](https://user-images.githubusercontent.com/63472358/205437801-08378c68-cd69-442f-9e3d-82b9141262dc.png)

**Reply by joba-1** [reply]
so these are definitely no esp8266 boot messages. There is sth on the line killing the signal or keeping the esp from booting.

Power issues normaly start only after boot when the esp tries to start wifi: unlikely

**Comment by sfromis** [comment]
The PCB has two big chips, one of which may be using serial to communicate with the ESP8266. To flash, you may very well have to disable that chip to get the serial lines free for flashing. If you manage to get the ESP8266 flashed, next step would be to find out how you can control the device via the other MCU. It may be a Tuya MCU, which is normally supported by Tasmota. Or it may be something else.

**Comment by mojtabavi** [comment]
> The PCB has two big chips, one of which may be using serial to communicate with the ESP8266. To flash, you may very well have to disable that chip to get the serial lines free for flashing. If you manage to get the ESP8266 flashed, next step would be to find out how you can control the device via the other MCU. It may be a Tuya MCU, which is normally supported by Tasmota. Or it may be something else.

yes it is tuya mcu

**Comment by sfromis** [comment]
Then find out how you can disable it during flashing. Many of those chips have some sort of enable or reset pin.

**Reply by mojtabavi** [reply]
> Then find out how you can disable it during flashing. Many of those chips have some sort of enable or reset pin.

as I understand the MCU Chip is a SONiX SN8F57084
![Screenshot 2022-12-03 142135](https://user-images.githubusercontent.com/63472358/205437560-9cd6c4e3-390f-4643-9658-29857f03240e.png)

then I connect RST pin (pin4) to the ground and nothing changed!!!

**Reply by mojtabavi** [reply]
is the problem due to the lack of power supply of the programmer?

**Reply by sfromis** [reply]
The USB/serial adapter you showed earlier is what we call the "golden CH340G", recommended for having a reasonably good ability to supply power, 150 mA of 3.3V power, from the small LDO (U1 chip) near the voltage selection jumper. While I can't say anything for sure, my expectation is that this would normally be enough, but I don't know how much power the rest of the board may draw.

**Reply by joba-1** [reply]
The esp8266 has strapping pins that need to have the correct hi or lo level to boot. Not sure which of them are accessible here. You could check them and fix with a pullup or pulldown resistor. Use a low resistance like 1k, 470Ohm to override what might be on board but don’t just short.

**Comment by mojtabavi** [comment, marked as answer]
Surprisingly, by changing the programmer instead of using CH340G, the problem was solved
